mlrun/__main__.py

```diff
@@ -579,34 +588,19 @@ def project(context, name, url, run, arguments, artifact_path,
 
 
 @main.command()
-@click.option('--api', help='api and db service path/url')
-@click.option('--namespace', '-n', help='kubernetes namespace')
-@click.option('--pending', '-p', is_flag=True,
-              help='clean pending pods as well')
-@click.option('--running', '-r', is_flag=True,
-              help='clean running pods as well')
-def clean(api, namespace, pending, running):
-    """Clean completed or failed pods/jobs"""
-    k8s = K8sHelper(namespace)
-    #mldb = get_run_db(db or mlconf.dbpath).connect()
-    items = k8s.list_pods(namespace)
-    states = ['Succeeded', 'Failed']
-    if pending:
-        states.append('Pending')
-    if running:
-        states.append('Running')
-    print('{:10} {:16} {:8} {}'.format('state', 'started', 'type', 'name'))
-    for i in items:
-        task = i.metadata.labels.get('mlrun/class', '')
-        state = i.status.phase
-        # todo: clean mpi, spark, .. jobs (+CRDs)
-        if task and task in ['build', 'job', 'dask'] and state in states:
-            name = i.metadata.name
-            start = ''
-            if i.status.start_time:
-                start = i.status.start_time.strftime("%b %d %H:%M:%S")
-            print('{:10} {:16} {:8} {}'.format(state, start, task, name))
-            k8s.del_pod(name)
+@click.argument('kind', type=str, default='', required=False)
+@click.option('--api', help='api and db service url')
+@click.option('--label-selector', '-ls', default='', help='label selector')
+@click.option('--force', '-f', is_flag=True,
+              help='clean resources in transient states as well')
```

Comment:
Talked offline
Decided to give ability to the user to clean a specific object (job/function) resources, which will also enable to tackle scenarios like - a function that all of its pods stuck in pending state because of wrong image name